src/main/java/school/hei/haapi/service/AwardedCourseService.java

```diff
@@ -87,4 +87,33 @@ public List<AwardedCourse> getByCriteria(
         PageRequest.of(page.getValue() - 1, pageSize.getValue(), Sort.by(ASC, "name"));
     return awardedCourseDao.findByCriteria(teacherId, courseId, pageable);
   }
+
+  public List<AwardedCourse> getByTeacherId(
+      String teacherId, PageFromOne page, BoundedPageSize pageSize) {
+    Pageable pageable =
+        PageRequest.of(page.getValue() - 1, pageSize.getValue(), Sort.by(ASC, "course"));
+    return awardedCourseRepository.findByTeacherId(teacherId, pageable);
+  }
+
+  @Transactional
+  public List<AwardedCourse> createOrUpdateAwardedCoursesByTeacherId(
+      String teacherId, List<CreateAwardedCourse> createAwardedCourses) {
+    return createAwardedCourses.stream()
+        .map(
+            createAwardedCourse ->
+                createOrUpdateAwardedCourseByTeacherId(teacherId, createAwardedCourse))
+        .collect(toList());
+  }
```

Comment:
Should be mapped in a mapper object like AwardedCourseMapper, don't map in service

Reply:
This method can be easily implemented without any other method